scripts/cherry-pick-patch-commits.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Identifies and cherry-picks commits that are not features and do not contain breaking changes.
+ *
+ * Defaults to operating against the latest non-pre-release tag, but a tag can be specified via command-line argument.
+ * This will automatically attempt to cherry-pick appropriate commits since the tag, but will abort and skip any
+ * cherry-picks that result in conflicts.
+ *
+ * Note that this does not create a branch - you will be in detached HEAD state. You can create a branch afterwards
+ * if you want, via `git checkout -b <branchname>`.
+ */
+
+const args = process.argv.slice(2);
+
+const parser = require('conventional-commits-parser');
+const parserOpts = require('conventional-changelog-angular/parser-opts');
+const simpleGit = require('simple-git/promise')();
+const {spawnSync} = require('child_process');
+
+const CONFLICT_MESSAGE = 'after resolving the conflicts, mark the corrected paths';
+
+// Resolves to the most recent non-pre-release git tag.
+async function getMostRecentTag() {
+  const tags = await simpleGit.tags();
+  // Filter old independent-versioned tags and pre-releases
+  const filteredTags = tags.all.filter((tag) => /^v/.test(tag) && !/-/.test(tag));
+  return filteredTags[filteredTags.length - 1];
+}
+
+// Resolves to an array of commits after the given tag, from earliest to latest (for proper cherry-picking).
+async function getCommitsAfterTag(tag) {
+  if (!args.includes('--no-fetch')) {
+    await simpleGit.fetch();
+  }
+  const log = await simpleGit.log({
+    from: tag,
+    to: 'origin/master',
+    // simple-git defaults to only including subject + tag/branch (%s%d) in message, which adds noise and is not useful
+    // for finding BREAKING CHANGEs (which are in body, %b).
+    format: {
+      hash: '%H', // Contains full hash only, for cherry-picking
+      message: '%s\n\n%b', // Contains subject + body for passing to conventional-commits-parser
+      oneline: '%h %s' // Contains abbreviated hash + subject for summary lists
+    },
+    // Use a splitter that is very unlikely to be included in commit descriptions
+    splitter: ';;;'
```

Comment:
Where does this splitter get removed or modified, or is it coming from one of the git commands?

Reply:
It's used within `simple-git`. It defaults to `;` which I figure is risky if I'm including the body of commit messages in what gets split.

Reply:
See https://github.com/steveukx/git-js/blob/master/src/git.js#L1228

Reply:
Yeah, good call. That’s a safer choice.